sdk/eventhubs/azure_messaging_eventhubs/benches/benchmarks.rs

```diff
@@ -71,13 +71,97 @@ fn send_batch_benchmark(c: &mut Criterion) {
+        return;
+    }
+
+    let rt = Runtime::new().unwrap();
+
+    let namespace = env::var("EVENTHUBS_HOST")
+        .expect("Set EVENTHUBS_HOST in env (e.g. my-namespace.servicebus.windows.net)");
+    let event_hub = env::var("EVENTHUB_NAME").expect("Set EVENTHUB_NAME in env");
+
+    let credential =
+        DefaultAzureCredential::new().expect("Failed to create DefaultAzureCredential");
+
+    let client = rt.block_on(async move {
+        Arc::new(
+            ProducerClient::builder()
+                .open(namespace.as_str(), event_hub.as_str(), credential)
+                .await
+                .expect("Failed to create ProducerClient"),
+        )
+    });
+
+    let client = client.clone();
+    c.bench_function("eventhubs_get_properties", |b| {
+        let client = client.clone();
+        b.to_async(&rt).iter(move || {
+            let client = client.clone();
+            async move {
+                let _ = client
+                    .get_eventhub_properties()
+                    .await
+                    .expect("Failed to get event hub properties");
+            }
+        });
+    });
+}
+
+fn get_eventhub_partition_properties(c: &mut Criterion) {
+    setup();
+
+    // Check if the environment variable is set thus allowing the benchmarks to run
+    if azure_core_test::TestMode::current().unwrap_or_default() != azure_core_test::TestMode::Live {
+        println!("Skipping benchmarks. Set AZURE_TEST_MODE to run.");
+        return;
+    }
+
+    let rt = Runtime::new().unwrap();
+
+    let namespace = env::var("EVENTHUBS_HOST")
+        .expect("Set EVENTHUBS_HOST in env (e.g. my-namespace.servicebus.windows.net)");
+    let event_hub = env::var("EVENTHUB_NAME").expect("Set EVENTHUB_NAME in env");
+
+    let credential =
+        DefaultAzureCredential::new().expect("Failed to create DefaultAzureCredential");
+
+    let client = rt.block_on(async move {
+        Arc::new(
+            ProducerClient::builder()
+                .open(namespace.as_str(), event_hub.as_str(), credential)
+                .await
+                .expect("Failed to create ProducerClient"),
+        )
+    });
+
+    let client = client.clone();
+    c.bench_function("eventhubs_get_partition_properties", |b| {
+        let client = client.clone();
+        b.to_async(&rt).iter(move || {
+            let client = client.clone();
+            async move {
+                let _ = client
+                    .get_partition_properties("0")
+                    .await
+                    .expect("Failed to get event hub properties");
+            }
+        });
+    });
+}
+
 criterion_group!(
     name = send_batch_benchmarks;
     config = Criterion::default()
         .sample_size(100)
         .warm_up_time(std::time::Duration::new(1, 0))
-        .measurement_time(std::time::Duration::new(250, 0));
-    targets = send_batch_benchmark
+        .measurement_time(std::time::Duration::new(500, 0));
```

Comment:
250 vs 500 seems a bit magical other than one being longer than the other. Is there a benefit for longer runs?

Reply:
250 causes criterion to complain that it is too low.
500 silences the criterion warning (technically anything over 450).

It appears that "measurement_time" is a max time for the test to run (250 seconds vs 500 seconds). But in practice the test runs much faster.